#!/usr/bin/env python
PACKAGE = "depthai_examples"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#gen.add("capture_still",   bool_t,   0, "Capture a still image",  False)

#ToDo: Add enums for modes!

gen.add("auto_focus_mode",    int_t,    0, "AutoFocusMode (0=Off, 1=AUTO, 2=MACRO, 3=VIDEO, 4=PIC, 5=EDOF)", 1,  0, 5)
gen.add("manual_focus",    int_t,    0, "Specify manual focus/lens position", 128,  0, 255)

gen.add("auto_exposure_lock",   bool_t,   0, "Auto exposure lock mode enabled or disabled",  False)
gen.add("auto_exposure_compensation",    int_t,    0, "Specify auto exposure compensation", 0,  -9, +9)
gen.add("anti_banding_mode",    int_t,    0, "AntiBandingMode (0=Off, 1=MAINS_50_HZ, 2=MAINS_60_HZ, 3=AUTO)", 3,  0, 3)
gen.add("manual_exposure_time",    int_t,    0, "Exposure time in microseconds. 0 for AutoExpose", 0,  0, 10000)
gen.add("manual_exposure_iso",    int_t,    0, "Sensitivity as ISO value", 100,  100, 1600)

gen.add("auto_white_balance_mode",    int_t,    0, "AutoWhiteBalanceMode (0=Off, 1=AUTO, 2=INCANDESCENT, 3=FLUORESCENT, 4=WARM_FLUORESCENT, 5=DAYLIGHT, 6=CLOUDY_DAYLIGHT, 7=TWILIGHT, 8=SHADE)", 1,  0, 8)
gen.add("manual_white_balance",    int_t,    0, "Light source color temperature in kelvins, range 1000..12000", 8000,  1000, 12000)

exit(gen.generate(PACKAGE, "depthai_examples", "CameraControl"))

"""
   enum class AutoFocusMode : uint8_t {
        /**
         * Autofocus disabled. Suitable for manual focus
         */
        OFF = 0,
        /**
         * Basic automatic focus mode. In this mode, the lens does not move unless the autofocus trigger action is called.
         */
        AUTO,
        /**
         * Close-up focusing mode - this mode is optimized for focusing on objects very close to the camera.
         */
        MACRO,
        /**
         * In this mode, the AF algorithm modifies the lens position continually to attempt to provide a constantly-in-focus image stream.
         * The focusing behavior should be suitable for good quality video recording; typically this means slower focus movement and no overshoots.
         */
        CONTINUOUS_VIDEO,
        /**
         * In this mode, the AF algorithm modifies the lens position continually to attempt to provide a constantly-in-focus image stream.
         * The focusing behavior should be suitable for still image capture; typically this means focusing as fast as possible
         */
        CONTINUOUS_PICTURE,
        /**
         * Extended depth of field (digital focus) mode. The camera device will produce images with an extended depth of field automatically.
         * AF triggers are ignored.
         */
        EDOF,
    };
"""

"""
    enum class AutoWhiteBalanceMode : uint8_t {
        /**
         * The camera device's auto-white balance routine is disabled.
         */
        OFF = 0,
        /**
         * The camera device's auto-white balance routine is active.
         */
        AUTO,
        /**
         * The camera device's auto-white balance routine is disabled; the camera device uses incandescent light as the assumed scene illumination for white
         * balance.
         */
        INCANDESCENT,
        /**
         * The camera device's auto-white balance routine is disabled; the camera device uses fluorescent light as the assumed scene illumination for white
         * balance.
         */
        FLUORESCENT,
        /**
         * The camera device's auto-white balance routine is disabled; the camera device uses warm fluorescent light as the assumed scene illumination for white
         * balance.
         */
        WARM_FLUORESCENT,
        /**
         * The camera device's auto-white balance routine is disabled; the camera device uses daylight light as the assumed scene illumination for white
         * balance.
         */
        DAYLIGHT,
        /**
         * The camera device's auto-white balance routine is disabled; the camera device uses cloudy daylight light as the assumed scene illumination for white
         * balance.
         */
        CLOUDY_DAYLIGHT,
        /**
         * The camera device's auto-white balance routine is disabled; the camera device uses twilight light as the assumed scene illumination for white
         * balance.
         */
        TWILIGHT,
        /**
         * The camera device's auto-white balance routine is disabled; the camera device uses shade light as the assumed scene illumination for white balance.
         */
        SHADE,
    };
"""